src/test/java/com/exasol/adapter/document/UdfCallBuilderTest.java

```diff
@@ -30,38 +28,115 @@ class UdfCallBuilderTest {
+        final RemoteTableQuery remoteTableQuery = getRemoteTableQueryWithOneColumn();
         final QueryPlan queryPlan = new EmptyQueryPlan();
         final String udfCallSql = UDF_CALL_BUILDER.getUdfCallSql(queryPlan, remoteTableQuery);
         assertThat(udfCallSql, equalTo("SELECT * FROM (VALUES (CAST(NULL AS  VARCHAR(123)))) WHERE FALSE"));
     }
 
+    @ParameterizedTest
+    @MethodSource("columnTypes")
+    void testBuildForEmptyPlanWithDataTypes(final ColumnMapping column, final String expectedCastType) {
+        final RemoteTableQuery remoteTableQuery = getRemoteTableQueryWithOneColumn(column);
+        final QueryPlan queryPlan = new EmptyQueryPlan();
+        final String udfCallSql = UDF_CALL_BUILDER.getUdfCallSql(queryPlan, remoteTableQuery);
+        assertThat(udfCallSql, equalTo("SELECT * FROM (VALUES (CAST(NULL AS  " + expectedCastType + "))) WHERE FALSE"));
+    }
+
+    static Stream<Arguments> columnTypes() {
+        final String colName = "TEST_COLUMN";
+        return Stream.of(
+                Arguments.of(PropertyToTimestampColumnMapping.builder().exasolColumnName(colName)
+                        .useTimestampWithLocalTimezoneType(true).build(), "TIMESTAMP WITH LOCAL TIME ZONE"),
+                Arguments.of(PropertyToTimestampColumnMapping.builder().exasolColumnName(colName)
+                        .useTimestampWithLocalTimezoneType(false).build(), "TIMESTAMP"),
+                Arguments.of(PropertyToDateColumnMapping.builder().exasolColumnName(colName).build(), "DATE"),
+                Arguments.of(
+                        PropertyToJsonColumnMapping.builder().exasolColumnName(colName).varcharColumnSize(5).build(),
+                        "VARCHAR(5)"),
+                Arguments.of(PropertyToBoolColumnMapping.builder().exasolColumnName(colName).build(), "BOOLEAN"),
+                Arguments.of(PropertyToDecimalColumnMapping.builder().exasolColumnName(colName).decimalPrecision(5)
+                        .decimalScale(3).build(), "DECIMAL(5,3)"),
+                Arguments.of(PropertyToDoubleColumnMapping.builder().exasolColumnName(colName).build(),
+                        "DOUBLE PRECISION"),
+                Arguments.of(
+                        PropertyToVarcharColumnMapping.builder().exasolColumnName(colName).varcharColumnSize(5).build(),
+                        "VARCHAR(5)"));
+    }
+
     @Test
     void testBasicSqlBuilding() {
-        final RemoteTableQuery remoteTableQuery = getRemoteTableQueryWithOneColumns();
+        final RemoteTableQuery remoteTableQuery = getRemoteTableQueryWithOneColumn();
+        final FetchQueryPlan queryPlan = new FetchQueryPlan(List.of(), new NoPredicate());
+        final String udfCallSql = UDF_CALL_BUILDER.getUdfCallSql(queryPlan, remoteTableQuery);
+        assertThat(udfCallSql, matchesRegex(quoteRegex(
+                "SELECT \"TEST_COLUMN\" FROM (SELECT \"ADAPTERS\".IMPORT_FROM_TEST_ADAPTER(\"DATA_LOADER\", '")
+                + "[^']+" + quoteRegex(
```

Comment:
This code is quite hard to read. I recommend using line breaks and indentation to make it more clear which parts are quoted and which are raw regex.